services/QuillLMS/engines/evidence/spec/factories/evidence/activity_healths.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :evidence_activity_health, class: 'Evidence::ActivityHealth' do
+    name { "test name" }
+    flag { "production" }
+    activity_id { 1 }
```

Comment:
I noticed this `activity_id` and then checked back on the models/evidence/activity_health.rb file and noticed there was no `belongs_to :activity`.  The schema says that it can be nil but is that what we want?

Sorry of for the backend question on a frontend PR but figured I would ask.

Reply:
So, because the EvidenceActivityHealth records are essentially just cached data for a simple front end display, I think they don't really need to be attached to Evidence Activities. This is the pattern we use in the other Activity Health dashboard for Connect and Grammar.

We do this because there's not actually any use cases where we would want to pull an EvidenceActivityHealth record from an activity itself. When we want to pull specific data for an activity, we actually just run the queries again fresh, since the data in these EvidenceActivityHealth is one day old or more.

These EvidenceActivityHealth records essentially just exist as an "out-of-date, but useful for a quick glance" thing for front-end use, but if admin want to do any deeper work with an activity's health data they will go to the Activity's actual dashboard and see the real-time data live.

Reply:
That makes sense.  Thank you for the thorough explanation and context.